src/components/Clusters/ClusterDetail.tsx

```diff
@@ -1,5 +1,140 @@
+import { Button, Card, CardSection, Group, Text } from '@mantine/core';
+import { useState } from 'react';
+import { Helmet } from 'react-helmet-async';
+import { RiDeleteBin5Line, RiSettings4Line, RiToggleFill, RiToggleLine } from 'react-icons/ri';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useAuth } from '@/hooks/useAuth';
+import { useClusterDetails, useJobsOnCluster } from '@/hooks/useClusters';
+import { disableModal, enableModal } from '@/utils/modals';
+import { pluralize } from '@/utils/string';
+
+import ClusterConfigCard from './ClusterConfigCard';
+import ClusterJobsCard from './ClusterJobsCard';
+import ClusterLabelBadge from './ClusterLabelBadge';
+import ClusterOwnerDetails from './ClusterOwnerDetails';
+
 function ClusterDetail() {
-  return <div className="flex h-full flex-col" />;
+  const navigate = useNavigate();
+  const { clusterId: clusterName = '' } = useParams();
+  if (!clusterName) navigate('/404', { replace: true });
+
+  const { data: cluster = null } = useClusterDetails(clusterName);
+  const [enabled, setEnabled] = useState(!cluster?.disabled);
+  const { user } = useAuth();
+
+  const { data: clusterJobs = [], isFetched } = useJobsOnCluster(clusterName);
+
+  const activeJobs = clusterJobs?.filter(
```

Comment:
Currently in this code, activeJobs would be undefined if clusterJobs is undefined. This would make the code break down further below where you are trying to do activeJob.length.

Instead do this,

const activeJobs = (clusterJobs ?? []).filter(...);

This way activeJobs will always be an array.

Reply:
I think clusterJobs will be empty array if it is undefined (`data: clusterJobs = []`).

Reply:
Ah missed that. Ok, then you don't need the ?.